### ISSUE TITLE
DAML SDK Docker image fails to start Sandbox with TLS certificates

### ISSUE THREAD

**Issue by SamirTalwar-DA** [digital-asset/daml#6965]
When attempting to start Sandbox in the DAML SDK 1.3.0 Docker image with a TLS certificate and PEM file specified, I get the following (very, very long) error:

```
~ $ daml sandbox --ledgerid=test --pem=/certs/server.pem --crt=/certs/server.crt --cacrt=/certs/ca.crt
INFO: Slf4jLogger started
Aug 03, 2020 1:05:57 PM io.grpc.netty.GrpcSslContexts defaultSslProvider
INFO: Java 9 ALPN API unavailable (this may be normal)
Aug 03, 2020 1:05:57 PM io.grpc.netty.GrpcSslContexts defaultSslProvider
INFO: netty-tcnative unavailable (this may be normal)
java.lang.IllegalArgumentException: Failed to load any of the given libraries: [netty_tcnative_linux_x86_64, netty_tcnative_linux_x86_64_fedora, netty_tcnative_x86_64, netty_tcnative]
        at io.netty.util.internal.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:104)
        at io.netty.handler.ssl.OpenSsl.loadTcNative(OpenSsl.java:581)
        at io.netty.handler.ssl.OpenSsl.<clinit>(OpenSsl.java:133)
        at io.grpc.netty.GrpcSslContexts.defaultSslProvider(GrpcSslContexts.java:217)
        at io.grpc.netty.GrpcSslContexts.configure(GrpcSslContexts.java:144)
        at io.grpc.netty.GrpcSslContexts.forServer(GrpcSslContexts.java:103)
        at com.daml.ledger.api.tls.TlsConfiguration.server(TlsConfiguration.scala:47)
        at com.daml.platform.apiserver.StandaloneApiServer.$anonfun$acquire$13(StandaloneApiServer.scala:120)
        at scala.Option.flatMap(Option.scala:271)
        at com.daml.platform.apiserver.StandaloneApiServer.$anonfun$acquire$12(StandaloneApiServer.scala:120)
        at com.daml.resources.ResourceOwner$$anon$2.$anonfun$acquire$1(ResourceOwner.scala:40)
        at scala.util.Success.$anonfun$map$1(Try.scala:255)
        at scala.util.Success.map(Try.scala:213)
        at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
        at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
        at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
        Suppressed: java.lang.UnsatisfiedLinkError: /tmp/libnetty_tcnative_linux_x86_645626458459689472423.so: Error loading shared library libcrypt.so.1: No such file or directory (needed by /tmp/libnetty_tcnative_linux_x86_645626458459689472423.so)
                at java.lang.ClassLoader$NativeLibrary.load(Native Method)
                at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1941)
                at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1824)
                at java.lang.Runtime.load0(Runtime.java:809)
                at java.lang.System.load(System.java:1086)
                at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:36)
                at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:351)
                at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:205)
                at io.netty.util.internal.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:96)
                ... 19 more
                Suppressed: java.lang.UnsatisfiedLinkError: /tmp/libnetty_tcnative_linux_x86_645626458459689472423.so: Error loading shared library libcrypt.so.1: No such file or directory (needed by /tmp/libnetty_tcnative_linux_x86_645626458459689472423.so)
                        at java.lang.ClassLoader$NativeLibrary.load(Native Method)
                        at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1941)
                        at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1824)
                        at java.lang.Runtime.load0(Runtime.java:809)
                        at java.lang.System.load(System.java:1086)
                        at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:36)
                        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
                        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                        at java.lang.reflect.Method.invoke(Method.java:498)
                        at io.netty.util.internal.NativeLibraryLoader$1.run(NativeLibraryLoader.java:371)
                        at java.security.AccessController.doPrivileged(Native Method)
                        at io.netty.util.internal.NativeLibraryLoader.loadLibraryByHelper(NativeLibraryLoader.java:363)
                        at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:341)
                        ... 21 more
                Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative_linux_x86_64 in java.library.path
                        at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1867)
                        at java.lang.Runtime.loadLibrary0(Runtime.java:870)
                        at java.lang.System.loadLibrary(System.java:1122)
                        at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
                        at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:351)
                        at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:136)
                        ... 20 more
                        Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative_linux_x86_64 in java.library.path
                                at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1867)
                                at java.lang.Runtime.loadLibrary0(Runtime.java:870)
                                at java.lang.System.loadLibrary(System.java:1122)
                                at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
                                at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                                at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
                                at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                                at java.lang.reflect.Method.invoke(Method.java:498)
                                at io.netty.util.internal.NativeLibraryLoader$1.run(NativeLibraryLoader.java:371)
                                at java.security.AccessController.doPrivileged(Native Method)
                                at io.netty.util.internal.NativeLibraryLoader.loadLibraryByHelper(NativeLibraryLoader.java:363)
                                at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:341)
                                ... 21 more
        Suppressed: java.lang.UnsatisfiedLinkError: could not load a native library: netty_tcnative_linux_x86_64_fedora
                at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:226)
                at io.netty.util.internal.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:96)
                ... 19 more
        Caused by: java.io.FileNotFoundException: META-INF/native/libnetty_tcnative_linux_x86_64_fedora.so
                at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:175)
                ... 20 more
                Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative_linux_x86_64_fedora in java.library.path
                        at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1867)
                        at java.lang.Runtime.loadLibrary0(Runtime.java:870)
                        at java.lang.System.loadLibrary(System.java:1122)
                        at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
                        at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:351)
                        at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:136)
                        ... 20 more
                        Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative_linux_x86_64_fedora in java.library.path
                                at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1867)
                                at java.lang.Runtime.loadLibrary0(Runtime.java:870)
                                at java.lang.System.loadLibrary(System.java:1122)
                                at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
                                at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                                at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
                                at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                                at java.lang.reflect.Method.invoke(Method.java:498)
                                at io.netty.util.internal.NativeLibraryLoader$1.run(NativeLibraryLoader.java:371)
                                at java.security.AccessController.doPrivileged(Native Method)
                                at io.netty.util.internal.NativeLibraryLoader.loadLibraryByHelper(NativeLibraryLoader.java:363)
                                at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:341)
                                ... 21 more
        Suppressed: java.lang.UnsatisfiedLinkError: could not load a native library: netty_tcnative_x86_64
                at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:226)
                at io.netty.util.internal.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:96)
                ... 19 more
        Caused by: java.io.FileNotFoundException: META-INF/native/libnetty_tcnative_x86_64.so
                at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:175)
                ... 20 more
                Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative_x86_64 in java.library.path
                        at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1867)
                        at java.lang.Runtime.loadLibrary0(Runtime.java:870)
                        at java.lang.System.loadLibrary(System.java:1122)
                        at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
                        at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:351)
                        at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:136)
                        ... 20 more
                        Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative_x86_64 in java.library.path
                                at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1867)
                                at java.lang.Runtime.loadLibrary0(Runtime.java:870)
                                at java.lang.System.loadLibrary(System.java:1122)
                                at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
                                at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                                at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
                                at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                                at java.lang.reflect.Method.invoke(Method.java:498)
                                at io.netty.util.internal.NativeLibraryLoader$1.run(NativeLibraryLoader.java:371)
                                at java.security.AccessController.doPrivileged(Native Method)
                                at io.netty.util.internal.NativeLibraryLoader.loadLibraryByHelper(NativeLibraryLoader.java:363)
                                at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:341)
                                ... 21 more
        Suppressed: java.lang.UnsatisfiedLinkError: could not load a native library: netty_tcnative
                at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:226)
                at io.netty.util.internal.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:96)
                ... 19 more
        Caused by: java.io.FileNotFoundException: META-INF/native/libnetty_tcnative.so
                at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:175)
                ... 20 more
                Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative in java.library.path
                        at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1867)
                        at java.lang.Runtime.loadLibrary0(Runtime.java:870)
                        at java.lang.System.loadLibrary(System.java:1122)
                        at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
                        at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:351)
                        at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:136)
                        ... 20 more
                        Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative in java.library.path
                                at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1867)
                                at java.lang.Runtime.loadLibrary0(Runtime.java:870)
                                at java.lang.System.loadLibrary(System.java:1122)
                                at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
                                at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                                at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
                                at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                                at java.lang.reflect.Method.invoke(Method.java:498)
                                at io.netty.util.internal.NativeLibraryLoader$1.run(NativeLibraryLoader.java:371)
                                at java.security.AccessController.doPrivileged(Native Method)
                                at io.netty.util.internal.NativeLibraryLoader.loadLibraryByHelper(NativeLibraryLoader.java:363)
                                at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:341)
                                ... 21 more

Aug 03, 2020 1:05:57 PM io.grpc.netty.GrpcSslContexts defaultSslProvider
INFO: Conscrypt not found (this may be normal)
java.lang.ClassNotFoundException: org.conscrypt.Conscrypt
        at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:264)
        at io.grpc.internal.ConscryptLoader.newProvider(ConscryptLoader.java:72)
        at io.grpc.netty.GrpcSslContexts$ConscryptHolder.<clinit>(GrpcSslContexts.java:277)
        at io.grpc.netty.GrpcSslContexts.findJdkProvider(GrpcSslContexts.java:250)
        at io.grpc.netty.GrpcSslContexts.defaultSslProvider(GrpcSslContexts.java:221)
        at io.grpc.netty.GrpcSslContexts.configure(GrpcSslContexts.java:144)
        at io.grpc.netty.GrpcSslContexts.forServer(GrpcSslContexts.java:103)
        at com.daml.ledger.api.tls.TlsConfiguration.server(TlsConfiguration.scala:47)
        at com.daml.platform.apiserver.StandaloneApiServer.$anonfun$acquire$13(StandaloneApiServer.scala:120)
        at scala.Option.flatMap(Option.scala:271)
        at com.daml.platform.apiserver.StandaloneApiServer.$anonfun$acquire$12(StandaloneApiServer.scala:120)
        at com.daml.resources.ResourceOwner$$anon$2.$anonfun$acquire$1(ResourceOwner.scala:40)
        at scala.util.Success.$anonfun$map$1(Try.scala:255)
        at scala.util.Success.map(Try.scala:213)
        at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
        at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
        at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

Aug 03, 2020 1:05:57 PM io.grpc.netty.GrpcSslContexts defaultSslProvider
INFO: Jetty ALPN unavailable (this may be normal)
java.lang.ClassNotFoundException: org/eclipse/jetty/alpn/ALPN
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:348)
        at io.grpc.netty.JettyTlsUtil.isJettyAlpnConfigured(JettyTlsUtil.java:64)
        at io.grpc.netty.GrpcSslContexts.findJdkProvider(GrpcSslContexts.java:241)
        at io.grpc.netty.GrpcSslContexts.defaultSslProvider(GrpcSslContexts.java:221)
        at io.grpc.netty.GrpcSslContexts.configure(GrpcSslContexts.java:144)
        at io.grpc.netty.GrpcSslContexts.forServer(GrpcSslContexts.java:103)
        at com.daml.ledger.api.tls.TlsConfiguration.server(TlsConfiguration.scala:47)
        at com.daml.platform.apiserver.StandaloneApiServer.$anonfun$acquire$13(StandaloneApiServer.scala:120)
        at scala.Option.flatMap(Option.scala:271)
        at com.daml.platform.apiserver.StandaloneApiServer.$anonfun$acquire$12(StandaloneApiServer.scala:120)
        at com.daml.resources.ResourceOwner$$anon$2.$anonfun$acquire$1(ResourceOwner.scala:40)
        at scala.util.Success.$anonfun$map$1(Try.scala:255)
        at scala.util.Success.map(Try.scala:213)
        at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
        at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
        at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

ERROR: Shutting down because of an initialization error.
java.lang.IllegalStateException: Could not find TLS ALPN provider; no working netty-tcnative, Conscrypt, or Jetty NPN/ALPN available
        at io.grpc.netty.GrpcSslContexts.defaultSslProvider(GrpcSslContexts.java:233)
        at io.grpc.netty.GrpcSslContexts.configure(GrpcSslContexts.java:144)
        at io.grpc.netty.GrpcSslContexts.forServer(GrpcSslContexts.java:103)
        at com.daml.ledger.api.tls.TlsConfiguration.server(TlsConfiguration.scala:47)
        at com.daml.platform.apiserver.StandaloneApiServer.$anonfun$acquire$13(StandaloneApiServer.scala:120)
        at scala.Option.flatMap(Option.scala:271)
        at com.daml.platform.apiserver.StandaloneApiServer.$anonfun$acquire$12(StandaloneApiServer.scala:120)
        at com.daml.resources.ResourceOwner$$anon$2.$anonfun$acquire$1(ResourceOwner.scala:40)
        at scala.util.Success.$anonfun$map$1(Try.scala:255)
        at scala.util.Success.map(Try.scala:213)
        at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
        at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
        at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
daml-helper: Received ExitFailure 1 when running
Raw command: java -Dlogback.configurationFile=/home/daml/.daml/sdk/1.3.0/daml-sdk/sandbox-logback.xml -jar /home/daml/.daml/sdk/1.3.0/daml-sdk/daml-sdk.jar sandbox --ledgerid=test --pem=/certs/server.pem --crt=/certs/server.crt --cacrt=/certs/ca.crt
```

This basically boils down to _netty-tcnative_ failing to load any of the supported TLS providers (of which at least OpenSSL is available). This is because OpenSSL relies on dynamically linking against _libc_, which doesn't exist in Alpine Linux, because it favors static linking with _musl_ instead of _glibc_.

This can be fixed by installing `libc6-compat` to provide a dynamic linking compatibility layer. I believe this still interfaces with _musl_.

We could also stop using Alpine as a base image.

**Comment by SamirTalwar-DA** [comment]
You can create some certificates and keys to test this as follows:

```
openssl genrsa -out ca.key 4096
openssl req -new -x509 -key ca.key -out ca.crt -subj '/CN=0.0.0.0.ca' -days 3650

openssl genrsa -out server.key 4096
openssl pkey -in server.key -out server.pem
openssl req -new -key server.key -out server.csr -subj '/CN=0.0.0.0.server'
openssl x509 -req -in server.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out server.crt -extfile /path/to/digital-asset/daml/ledger/test-common/test-certificates/openssl-extensions.cnf -extensions req_ext -days 3650

openssl genrsa -out client.key 4096
openssl pkey -in client.key -out client.pem
openssl req -new -key client.key -out client.csr -subj '/CN=0.0.0.0.client'
openssl x509 -req -in client.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out client.crt -extfile /path/to/digital-asset/daml/ledger/test-common/test-certificates/openssl-extensions.cnf -extensions req_ext -days 3650
```

**Comment by garyverhaegen-da** [comment]
Fixed by #6970.
